core/math/basis.cpp

```diff
@@ -1049,9 +1049,10 @@ Basis Basis::looking_at(const Vector3 &p_target, const Vector3 &p_up, bool p_use
 		v_z = -v_z;
 	}
 	Vector3 v_x = p_up.cross(v_z);
-#ifdef MATH_CHECKS
-	ERR_FAIL_COND_V_MSG(v_x.is_zero_approx(), Basis(), "The target vector and up vector can't be parallel to each other.");
-#endif
+	if (v_x.is_zero_approx()) {
+		WARN_PRINT("Target and up vectors are colinear. This is not advised as it may cause unwanted rotation around local Z axis.");
```

Comment:
This might spam a lot, shouldn't it be `WARN_PRINT_ONCE`?

In general this kind of warnings aren't super helpful to end users, as a basis may end up in this state from internal engine code or complex calculations, and the warning doesn't point at which basis / object transform is referred to. So I'm not convinced this will make end users happy.

Reply:
If look_at is called continuously, it may be spammed, but in that case it should not be called continuously in the first place, so spamming it is not considered excessive.

As mentioned in https://github.com/godotengine/godot/pull/100991#issuecomment-2569178136, if the orientation is nearly the same, the user should detect it and do nothing. Well, this could be add documentation or changing a warning text.